### DISCUSSION TITLE
MySQL custom config example and RAM consumption improvement

### DISCUSSION THREAD

**Discussion by Jaimer** [caprover/caprover#957] [Show and tell]
I just wanted to share a solution for a RAM consumption issue I was having.

After installing the one click app for Wordpress + MySQL i noticed an increase of RAM consumption on my server reaching almost 80%. I found some [tweaks](https://stackoverflow.com/questions/10676753/reducing-memory-consumption-of-mysql-on-ubuntuaws-micro-instance) on the MySQL configuration that could help but this needed to be included on the my.cnf file which would get rewriten after the container restarts.

So the solution is to write a custom my.cnf and a custom dockerfile to use the base image of MySQL but copy the custom my.cnf into the image at build time and finally the captain-definition file to deploy via CLI or Web.

I managed to reduce my RAM usage to 60% from 80%.

I leave [here](https://github.com/Jaimer/MySQL-caprover-custom-config) the example of my captain-definition, dockerfile and my.cnf files.

This same procedure can be done for any other image to customize its configuration.

**Comment by githubsaturn** [comment]
Do you have a high amount of traffic for your wordpress site?

If not, then the default configs should work just fine. Checkout the [demo instance](https://captain.server.demo.caprover.com/) (password: captain42) where a single instance of wordpress almost consumes **1% CPU and only  50mb of RAM**.

**Reply by Jaimer** [reply]
I don't know why, but just after I installed one instance of Wordpress my RAM usage elevated considerably. It's a new site and should have almost 0 traffic. But was seeing ~260MB  usage for the db and ~260MB for the WP container.

![image](https://user-images.githubusercontent.com/447088/103490746-e89a1f00-4deb-11eb-9fa4-c592c9da0b3c.png)

I'm going to create a new site and watch the ram usage, maybe it has something to do with the installed plugins.

**Reply by githubsaturn** [reply]
Yes, it's probably because of some plugins that you're using.

**Comment by githubsaturn** [comment]
![image](https://user-images.githubusercontent.com/10582082/103482790-dea6fa80-4db0-11eb-8919-50c6e712854c.png)
